### PR TITLE
Learner id refactor

### PR DESCRIPTION
**Pull request by zhengp0**

### CODE REVIEW

File: src/modrover/strategies/backward_explore.py [hunk 1]
```diff
@@ -6,7 +6,7 @@ class BackwardExplore(RoverStrategy):
 
     def __init__(self, num_covariates: int):
         super().__init__(num_covariates)
-        self.base_learnerid = LearnerID(tuple(range(num_covariates + 1)))
+        self.base_learner_id = LearnerID(tuple(range(num_covariates + 1)))
```
Comment (davidshaw2018) [inline src/modrover/strategies/backward_explore.py:9]: probably don't need the LearnerID constructor anymore right?

File: tests/test_learnerid.py [hunk 2]
```diff
@@ -1,60 +0,0 @@
-from contextlib import nullcontext as does_not_raise
```
Comment (davidshaw2018) [inline tests/test_learnerid.py:1]: we might consider moving these tests to `test_strategy.py` instead of deleting altogether

**Review by davidshaw2018 — APPROVED** [review]

File: src/modrover/strategies/backward_explore.py (hunk 1 above)
Comment (zhengp0) [reply src/modrover/strategies/backward_explore.py:9]: Yes! that is a good point! Will change this

File: tests/test_learnerid.py (hunk 2 above)
Comment (zhengp0) [reply tests/test_learnerid.py:1]: That sounds good! I was too focus on making the tests pass :)

**Pull request merged**